redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java

```diff
@@ -83,7 +84,8 @@ public GridType overrideType(Type type) {
 
 	@Override
 	public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
-		Entity entity = getEntity( key );
+		Entity entity = entityStorageStrategy.getEntity( entityId( key ) );
```

Comment:
Yes, right. I've filed an issue to clarify it in the JavaDocs, the different dialects handle it not consistently atm.

On a related note, I am also thinking about making `EntityKey` an interface and let dialects return specific implementations. That way we may avoid some copying between `EntityKey` and the backend-specific structures.